test/hocon_schema_tests.erl

```diff
@@ -201,6 +201,16 @@ richmap_to_map_test_() ->
                       #{<<"b">> => [1, 2, #{<<"x">> => <<"foo">>}]}}, F("a.b = [1,2,{x=foo}]"))
     ].
 
+richmap_to_map_atom_key_test() ->
+    Sc = #{structs => [?VIRTUAL_ROOT],
+           fields => [ {metadata, hoconsc:t(string())}
+                     , {type, hoconsc:t(string())}
+                     , {value, hoconsc:t(string())}
+                     ]},
+    StrConf = "type=t, metadata=m, value=v",
+    {ok, M} = hocon:binary(StrConf, #{format => richmap}),
+    ?assertMatch(#{metadata := "m", type := "t", value := "v"},
```

Comment:
This clashes the `value` field in richmap.
To keep richmap_to_map maive, maybe we need to use a special tag such as `$value` for richmap.